packages/notifi-react-card/lib/components/subscription/subscription-card-views/HistoryCardView.tsx

```diff
@@ -176,6 +192,22 @@ export const AlertHistoryView: React.FC<AlertHistoryViewProps> = ({
     }
   }, [client]);
 
+  const validateSupportedDetailType = useCallback(
+    (entry?: NotificationHistoryEntry): boolean => {
+      const supportedEntryDetailTypes = [
+        'BroadcastMessageEventDetails',
+        'HealthValueOverThresholdEventDetails',
+        'GenericEventDetails',
+        'ChatMessageReceivedEventDetails',
+        'AccountBalanceChangedEventDetails',
+      ];
+      if (supportedEntryDetailTypes.includes(entry?.detail?.__typename ?? ''))
+        return true;
+      return false;
+    },
+    [],
```

Comment:
a `useCallback` with no dependencies can simply be a module-level function -- let's move the function declaration outside the component (just moving it to line 137 should be sufficient)

Reply:
Refactored quit a bit, all related logics are migrated to useNotificationHistory hook.